ssl/statem/statem_lib.c

```diff
@@ -1039,14 +1039,25 @@ WORK_STATE tls_finish_handshake(SSL *s, WORK_STATE wst, int clearbufs, int stop)
     int cleanuphand = s->statem.cleanuphand;
 
     if (clearbufs) {
-        if (!SSL_IS_DTLS(s)) {
+        if (!SSL_IS_DTLS(s)
+#ifndef OPENSSL_NO_SCTP
+                /*
+                 * RFC6083: SCTP provides a reliable and in-sequence transport service for DTLS
+                 * messages that require it. Therefore, DTLS procedures for retransmissions
+                 * MUST NOT be used.
+                 * Hence the init_buf can be cleared when DTLS over SCTP as transport is used.
+                 */
```

Comment:
The comment block is indented too far by four bytes. But never mind, I will fix it when I merge your pull request.